internal/controller/controller.go

```diff
@@ -507,18 +509,68 @@ func (c *Controller) deployConfiguration(config *dynamic.Configuration) error {
 		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
 	}
 
-	for _, pod := range podList.Items {
-		log.Debugf("Deploying to pod %s with IP %s", pod.Name, pod.Status.PodIP)
+	if err := deployToPods(podList.Items, config); err != nil {
+		return fmt.Errorf("error deploying configuration: %v", err)
+	}
 
-		if deployErr := c.deployToPod(pod.Name, pod.Status.PodIP, config); deployErr != nil {
-			return fmt.Errorf("error deploying configuration: %v", deployErr)
+	return nil
+}
+
+// deployConfigurationToUnreadyNodes deploys the configuration to the mesh pods.
+func (c *Controller) deployConfigurationToUnreadyNodes(config *dynamic.Configuration) error {
+	podList, err := c.clients.ListPodWithOptions(c.meshNamespace, metav1.ListOptions{
+		LabelSelector: "component==maesh-mesh",
+	})
+	if err != nil {
+		return fmt.Errorf("unable to retrieve pod list: %v", err)
+	}
+
+	if len(podList.Items) == 0 {
+		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
+	}
+
+	var unreadyPods []corev1.Pod
+
```

Comment:
```suggestion
```

Reply:
This makes the linter complain :/

Reply:
The linter is wrong here, it doesn't seem to understand that the unreadyPods is used in the nested for loop :( I'd suggest disabling this linter module or at least adding an exception for this because here I find this newline confusing :p

Reply:
I like having the newline, and agree with the linter. Although the var is used in the loop, it is not used in the iteration.